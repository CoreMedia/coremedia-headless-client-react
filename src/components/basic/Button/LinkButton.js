// @flow
import Button from './Button';

const LinkButton = Button.withComponent('a').extend`
  display: inline-block;
`;
LinkButton.displayName = 'LinkButton';

export default LinkButton;
