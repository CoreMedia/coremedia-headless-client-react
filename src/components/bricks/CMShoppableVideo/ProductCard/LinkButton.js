// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  ariaLabel?: string,
};

const LinkButton: ComponentType<Props> = styled.a.attrs({
  role: 'button',
  'aria-label': props => props.ariaLabel,
})`
  display: inline-block;
  width: 100%;
`;
LinkButton.displayName = 'LinkButton';

LinkButton.propTypes = {
  ariaLabel: PropTypes.string,
};

export default LinkButton;
