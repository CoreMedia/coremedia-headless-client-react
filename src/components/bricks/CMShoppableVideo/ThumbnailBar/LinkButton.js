// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  active: boolean,
  ariaLabel?: string,
};

const LinkButton: ComponentType<Props> = styled.a.attrs({
  role: 'button',
  'aria-label': props => props.ariaLabel,
})`
  display: inline-block;
  opacity: ${props => (props.active ? 1 : 0.4)};
  transition: opacity 0.3s linear 0.3s;
`;
LinkButton.displayName = 'LinkButton';

LinkButton.propTypes = {
  active: PropTypes.bool.isRequired,
  ariaLabel: PropTypes.string,
};

export default LinkButton;
