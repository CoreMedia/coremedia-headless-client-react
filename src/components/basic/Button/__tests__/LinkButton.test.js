// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import LinkButton from '../LinkButton';

describe('LinkButton Component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<LinkButton href="https://labs.coremedia.com">Content</LinkButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
