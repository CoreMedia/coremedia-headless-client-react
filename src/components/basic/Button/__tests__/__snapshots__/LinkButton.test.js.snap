// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton Component should render correctly 1`] = `
.c0 {
  font-weight: 400;
  border: 2px solid;
  padding: 0.5em 1em;
  background: inherit;
  text-align: center;
  cursor: pointer;
  display: inline-block;
}

<a
  className="c0"
  href="https://labs.coremedia.com"
>
  Content
</a>
`;
